nnvm/python/nnvm/testing/check_computation.py

```diff
@@ -0,0 +1,451 @@
+                nnvm.compiler.graph_attr.set_shape_inputs(backward_graph, shape)
+
+            backward_graph = backward_graph.apply('InferShape').apply('InferType')
+            shapes = backward_graph.json_attr('shape')
+            dtypes = backward_graph.json_attr('dtype')
+        except nnvm._base.NNVMError as err:
+            if np_backward is None and numerical_grads == "if_possible":
+                logging.warning("Won't check gradients because: %s", str(err).split('\n', 1)[0])
+                numerical_grads = False
+                backward_graph = None
+            else:
+                raise
+
+    main_graph = backward_graph if backward_graph is not None else forward_graph
+
+    if dump_graph:
+        print()
+        print(main_graph.ir(join_node_attrs=['shape', 'dtype']))
+
+    # Generate random data for inputs (including head_grads)
+
+    np_inputs = {}
+
+    for x in main_graph.symbol.list_input_variables():
+        x_name = x.attr('name')
+
+        low = -1.0
+        high = 1.0
+        if in_range is not None:
+            if isinstance(in_range, dict):
+                if x_name in in_range:
+                    low = in_range[x_name][0]
+                    high = in_range[x_name][1]
+            else:
+                low = in_range[0]
+                high = in_range[1]
+
+        x_node_id = main_graph.index.node_id(x_name)
+        x_shape = shapes[x_node_id]
+        x_dtype = dtypes[x_node_id]
+
+        if not isinstance(x_dtype, str):
+            x_dtype = TCODE_TO_DTYPE[x_dtype]
+
+        x_value = np.random.uniform(size=x_shape, low=low, high=high).astype(x_dtype)
+
+        np_inputs[x_name] = x_value
+
+    np_inputs_without_head_grads = {k: np_inputs[k] for k in np_inputs if k != 'head_grads'}
+
+    # Compute and compare the results
+    for target, ctx in ctx_list():
+        if (exclude_targets is not None and (target in exclude_targets or
+                                             str(target) in exclude_targets)) or\
+           (only_targets is not None and not (target in only_targets or
+                                              str(target) in only_targets)):
+            logging.info("Skipping target = %s, ctx = %s", target, ctx)
+            continue
+
+        logging.info("Checking computation on target = %s, ctx = %s", target, ctx)
+
+        debug_stage = None
+
+        try:
+            debug_stage = "compiling"
+            nnvm_res = None
+            main_function = graph_to_function(main_graph, target, ctx)
+            # nnvm_res contains the output and gradients (if they are needed)
+            nnvm_res = main_function(**np_inputs)
+
+            if np_forward is not None:
+                debug_stage = "checking forward computation"
+                logging.debug(debug_stage)
+
+                numpy_res = np_forward(**np_inputs_without_head_grads)
+                np.testing.assert_allclose(nnvm_res[0], numpy_res, atol=atol, rtol=rtol)
```

Comment:
In the context of above comment on np_forward, 
Pls check if the result from np_forward is a list and is yes, then compare all arrays in list.